packages/SidePanel/src/hooks/useOffsetScrollFooter.js

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import debounce from "lodash.debounce";
+
+const useOffsetScrollFooter = (height, ref) => {
+  const [offsetY, setOffsetY] = React.useState(0);
+  const [opacity, setOpacity] = React.useState(1);
+
+  const handleOffset = debounce(() => {
+    const top = ref.current.scrollTop;
+
+    setOpacity(1);
+    setOffsetY(-top);
+  }, 350);
+
+  const handleScrolling = () => {
+    setOpacity(0);
+    handleOffset();
+  };
+
+  React.useEffect(() => {
+    if (ref.current) {
```

Comment:
I tried this, but since ref.current is null at the beginning the effect will not get executed, passing ref.current allowed to listen for current and when change applies the effect.